c++: error: unrecognized command line option '--diag_suppress=esa_on_defaulted_function_ignored'

Cuda fans, any idea about this error:
```
c++: error: unrecognized command line option '--diag_suppress=esa_on_defaulted_function_ignored'
```

I tried a different gcc already....

Details:
```
Running with gitlab-runner  (c364eff5)
  on ascent_copa_setuid 47b77dfd
Using SetUID Shell executor...
Running on login1...
Fetching changes...
Removing kokkos.install/
Removing kokkos/
HEAD is now at 6e6c9eb try newer gcc
Checking out 6e6c9eb9 as ci-cuda...
Skipping Git submodules setup
Downloading artifacts for BuildKokkos Cuda (38102)...
About to register the Shell executor type...
About to register the Batch Runner...
Downloading artifacts from coordinator... ok        id=38102 responseStatus=200 OK token=xheXy1AZ
$ CI_PROJECT_DIR=${PWD}
$ module load cmake
$ module load cuda
$ module load gcc/6.4.0

Lmod is automatically replacing "xl/16.1.1-1" with "gcc/6.4.0".


Due to MODULEPATH changes, the following have been reloaded:
  1) spectrum-mpi/10.2.0.10-20181214

$ for i in ${BACKENDS}; do CMAKE_OPTS+=( -DCabana_ENABLE_${i}=ON ); done
$ j="$(grep -c processor /proc/cpuinfo 2>/dev/null)" || j=0; ((j++))
$ mkdir build && cd build && cmake -DCMAKE_PREFIX_PATH=${CI_PROJECT_DIR}/kokkos.install -DCabana_ENABLE_TESTING=ON -DCabana_ENABLE_Serial=OFF -DCabana_ENABLE_EXAMPLES=OFF ${CMAKE_OPTS[@]} .. && make -k -j${j} VERBOSE=1 && make test CTEST_OUTPUT_ON_FAILURE=1
-- The CXX compiler identification is GNU 6.4.0
-- Check for working CXX compiler: /sw/ascent/gcc/6.4.0/bin/c++
-- Check for working CXX compiler: /sw/ascent/gcc/6.4.0/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found KOKKOS: /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/kokkos.install  
-- The CUDA compiler identification is NVIDIA 9.2.148
-- Check for working CUDA compiler: /sw/ascent/cuda/9.2.148/bin/nvcc
-- Check for working CUDA compiler: /sw/ascent/cuda/9.2.148/bin/nvcc -- works
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Enable Devices: Cuda
-- Found Doxygen: /bin/doxygen (found version "1.8.5") found components:  doxygen missing components:  dot
-- Performing Test COMPILER_SUPPORTS_MARCH
-- Performing Test COMPILER_SUPPORTS_MARCH - Failed
-- Found Git: /bin/git (found version "1.8.3.1") 
-- Cabana Revision = '6e6c9eb9aa8964c7cb2856f472ce60829eb4aa2d'
-- Configuring done
-- Generating done
-- Build files have been written to: /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build
/autofs/nccsopen-svm1_sw/ascent/.swci/0-core/opt/spack/20180914/linux-rhel7-ppc64le/gcc-4.8.5/cmake-3.12.2-vltu2cerbdpbvzwpil2fvw4skvjmabfl/bin/cmake -H/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana -B/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build --check-build-system CMakeFiles/Makefile.cmake 0
/autofs/nccsopen-svm1_sw/ascent/.swci/0-core/opt/spack/20180914/linux-rhel7-ppc64le/gcc-4.8.5/cmake-3.12.2-vltu2cerbdpbvzwpil2fvw4skvjmabfl/bin/cmake -E cmake_progress_start /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/CMakeFiles /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
make -f core/src/CMakeFiles/cabanacore.dir/build.make core/src/CMakeFiles/cabanacore.dir/depend
make -f core/unit_test/CMakeFiles/cabana_core_gtest.dir/build.make core/unit_test/CMakeFiles/cabana_core_gtest.dir/depend
make[2]: Entering directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
cd /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build && /autofs/nccsopen-svm1_sw/ascent/.swci/0-core/opt/spack/20180914/linux-rhel7-ppc64le/gcc-4.8.5/cmake-3.12.2-vltu2cerbdpbvzwpil2fvw4skvjmabfl/bin/cmake -E cmake_depends "Unix Makefiles" /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/src /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/src /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/src/CMakeFiles/cabanacore.dir/DependInfo.cmake --color=
make[2]: Entering directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
cd /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build && /autofs/nccsopen-svm1_sw/ascent/.swci/0-core/opt/spack/20180914/linux-rhel7-ppc64le/gcc-4.8.5/cmake-3.12.2-vltu2cerbdpbvzwpil2fvw4skvjmabfl/bin/cmake -E cmake_depends "Unix Makefiles" /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/unit_test /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test/CMakeFiles/cabana_core_gtest.dir/DependInfo.cmake --color=
Dependee "/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test/CMakeFiles/cabana_core_gtest.dir/DependInfo.cmake" is newer than depender "/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test/CMakeFiles/cabana_core_gtest.dir/depend.internal".
Dependee "/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/src/CMakeFiles/cabanacore.dir/DependInfo.cmake" is newer than depender "/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/src/CMakeFiles/cabanacore.dir/depend.internal".
Dependee "/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test/CMakeFiles/cabana_core_gtest.dir/depend.internal".
Dependee "/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/src/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/src/CMakeFiles/cabanacore.dir/depend.internal".
Scanning dependencies of target cabanacore
make[2]: Leaving directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
make -f core/src/CMakeFiles/cabanacore.dir/build.make core/src/CMakeFiles/cabanacore.dir/build
make[2]: Entering directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
[  2%] Building CXX object core/src/CMakeFiles/cabanacore.dir/impl/Cabana_Version.cpp.o
cd /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/src && /sw/ascent/gcc/6.4.0/bin/c++   -I/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/src -I/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/src -isystem /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/kokkos.install/include  -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/cabanacore.dir/impl/Cabana_Version.cpp.o -c /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/src/impl/Cabana_Version.cpp
Scanning dependencies of target cabana_core_gtest
make[2]: Leaving directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
make -f core/unit_test/CMakeFiles/cabana_core_gtest.dir/build.make core/unit_test/CMakeFiles/cabana_core_gtest.dir/build
make[2]: Entering directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
c++: error: unrecognized command line option '-Xcudafe'
[  5%] Building CXX object core/unit_test/CMakeFiles/cabana_core_gtest.dir/__/__/gtest/gtest/gtest-all.cc.o
cd /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test && /sw/ascent/gcc/6.4.0/bin/c++   -I/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/gtest -I/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test -I/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/core/unit_test  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   -std=c++11 -o CMakeFiles/cabana_core_gtest.dir/__/__/gtest/gtest/gtest-all.cc.o -c /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/gtest/gtest/gtest-all.cc
c++: error: unrecognized command line option '--diag_suppress=esa_on_defaulted_function_ignored'
make[2]: *** [core/src/CMakeFiles/cabanacore.dir/impl/Cabana_Version.cpp.o] Error 1
make[2]: Target `core/src/CMakeFiles/cabanacore.dir/build' not remade because of errors.
make[2]: Leaving directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
make[1]: *** [core/src/CMakeFiles/cabanacore.dir/all] Error 2
[  7%] Linking CXX static library libcabana_core_gtest.a
cd /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test && /autofs/nccsopen-svm1_sw/ascent/.swci/0-core/opt/spack/20180914/linux-rhel7-ppc64le/gcc-4.8.5/cmake-3.12.2-vltu2cerbdpbvzwpil2fvw4skvjmabfl/bin/cmake -P CMakeFiles/cabana_core_gtest.dir/cmake_clean_target.cmake
cd /root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build/core/unit_test && /autofs/nccsopen-svm1_sw/ascent/.swci/0-core/opt/spack/20180914/linux-rhel7-ppc64le/gcc-4.8.5/cmake-3.12.2-vltu2cerbdpbvzwpil2fvw4skvjmabfl/bin/cmake -E cmake_link_script CMakeFiles/cabana_core_gtest.dir/link.txt --verbose=1
/bin/ar qc libcabana_core_gtest.a  CMakeFiles/cabana_core_gtest.dir/__/__/gtest/gtest/gtest-all.cc.o
/bin/ranlib libcabana_core_gtest.a
make[2]: Leaving directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
[  7%] Built target cabana_core_gtest
make[1]: Target `all' not remade because of errors.
make[1]: Leaving directory `/root/hpc-gitlab-runner/ecp-copa/cabana/builds/users/junghans/47b77dfd/0/ecp-copa/cabana/build'
make: *** [all] Error 2
make: Target `default_target' not remade because of errors.
ERROR: Job failed: exit status 2
```

This is the GPU type:
```
jgalarowicz@login1.ascent ~]$ lspci -nn | grep '\[03'
0002:02:00.0 VGA compatible controller [0300]: ASPEED Technology, Inc. ASPEED Graphics Family [1a03:2000] (rev 41)
0004:04:00.0 3D controller [0302]: NVIDIA Corporation GV100GL [Tesla V100 SXM2] [10de:1db1] (rev a1)
0004:05:00.0 3D controller [0302]: NVIDIA Corporation GV100GL [Tesla V100 SXM2] [10de:1db1] (rev a1)
0035:03:00.0 3D controller [0302]: NVIDIA Corporation GV100GL [Tesla V100 SXM2] [10de:1db1] (rev a1)
0035:04:00.0 3D controller [0302]: NVIDIA Corporation GV100GL [Tesla V100 SXM2] [10de:1db1] (rev a1)
```
I just redid the builds with --with-cuda-options=enable_lambda, but the errors are still there.  I'll put a bigger listing here.  Maybe there will be more clues: 
```
[ 70%] Building CXX object core/unit_test/CMakeFiles/CartesianGrid_test.dir/tstCartesianGrid.cpp.o
[ 70%] Building CXX object core/unit_test/CMakeFiles/DeepCopy_test_Cuda.dir/Cuda/tstDeepCopy_Cuda.cpp.o
[ 70%] Building CXX object core/unit_test/CMakeFiles/CartesianGrid_test.dir/unit_test_main.cpp.o
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/SoA_test.dir/tstSoA.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/tstSoA.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/DeepCopy_test_Cuda.dir/Cuda/tstDeepCopy_Cuda.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/Cuda/tstDeepCopy_Cuda.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/AoSoA_test_Cuda.dir/unit_test_main.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/unit_test_main.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/Parallel_test_Cuda.dir/unit_test_main.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/unit_test_main.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/CartesianGrid_test.dir/tstCartesianGrid.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/tstCartesianGrid.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/Sort_test_Cuda.dir/Cuda/tstSort_Cuda.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/Cuda/tstSort_Cuda.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/Index_test.dir/tstIndex.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/tstIndex.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/DeepCopy_test_Cuda.dir/unit_test_main.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/unit_test_main.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/CartesianGrid_test.dir/unit_test_main.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/unit_test_main.cpp
cd /ccsopen/home/jgalarowicz/cabana/build/core/unit_test && /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper   -I/ccsopen/home/jgalarowicz/cabana/gtest -I/ccsopen/home/jgalarowicz/cabana/build/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/unit_test -I/ccsopen/home/jgalarowicz/cabana/core/src -I/ccsopen/home/jgalarowicz/cabana/build/core/src -isystem /ccsopen/home/jgalarowicz/kokkos.install/include  -DGTEST_HAS_PTHREAD=0 -O3 -DNDEBUG   --std=c++11 -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -o CMakeFiles/Sort_test_Cuda.dir/unit_test_main.cpp.o -c /ccsopen/home/jgalarowicz/cabana/core/unit_test/unit_test_main.cpp
/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

2 errors detected in the compilation of "/tmp/tmpxft_00006c91_00000000-6_tstTuple_Cuda.cpp1.ii".
make[2]: *** [core/unit_test/CMakeFiles/Tuple_test_Cuda.dir/Cuda/tstTuple_Cuda.cpp.o] Error 1
/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(685): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::TeamPolicy< ::Kokkos::Cuda,  ::Kokkos::Schedule< ::Kokkos::Dynamic>  > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(685): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(685): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::TeamPolicy< ::Kokkos::Cuda,  ::Kokkos::Schedule< ::Kokkos::Dynamic>  > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(685): error: identifier " const" is undefined in device code

4 errors detected in the compilation of "/tmp/tmpxft_00006c75_00000000-6_tstParallel_Cuda.cpp1.ii".
make[2]: *** [core/unit_test/CMakeFiles/Parallel_test_Cuda.dir/Cuda/tstParallel_Cuda.cpp.o] Error 1
/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2018): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelScan< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::initial const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2018): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2074): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelScan< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::final const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2074): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2087): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelScan< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::final const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2087): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

14 errors detected in the compilation of "/tmp/tmpxft_00006c99_00000000-6_tstLinkedCellList_Cuda.cpp1.ii".
make[2]: *** [core/unit_test/CMakeFiles/LinkedCellList_test_Cuda.dir/Cuda/tstLinkedCellList_Cuda.cpp.o] Error 1
/ccsopen/home/jgalarowicz/cabana/core/src/Cabana_Parallel.hpp(37): warning: calling a __host__ function from a __host__ __device__ function is not allowed
          detected during:
            instantiation of "std::enable_if<std::is_same<WorkTag, void>::value, void>::type Cabana::Impl::functorTagDispatch<WorkTag,FunctorType>(const FunctorType &, std::size_t, int) [with WorkTag=void, FunctorType=lambda [](int, int)->void]" 
(202): here
            instantiation of "void Cabana::neighbor_parallel_for(const Kokkos::RangePolicy<ExecParameters...> &, const FunctorType &, const NeighborListType &, const Cabana::SerialNeighborOpTag &, const std::__cxx11::string &) [with FunctorType=lambda [](int, int)->void, NeighborListType=Cabana::VerletList<Kokkos::CudaUVMSpace, Cabana::FullNeighborTag>, ExecParameters=<Kokkos::Cuda>]" 
/ccsopen/home/jgalarowicz/cabana/core/unit_test/tstNeighborList.hpp(381): here

/ccsopen/home/jgalarowicz/cabana/core/src/Cabana_Parallel.hpp(37): warning: calling a __host__ function from a __host__ __device__ function is not allowed
          detected during:
            instantiation of "std::enable_if<std::is_same<WorkTag, void>::value, void>::type Cabana::Impl::functorTagDispatch<WorkTag,FunctorType>(const FunctorType &, std::size_t, int) [with WorkTag=void, FunctorType=lambda [](int, int)->void]" 
(283): here
            instantiation of "void Cabana::neighbor_parallel_for(const Kokkos::RangePolicy<ExecParameters...> &, const FunctorType &, const NeighborListType &, const Cabana::TeamNeighborOpTag &, const std::__cxx11::string &) [with FunctorType=lambda [](int, int)->void, NeighborListType=Cabana::VerletList<Kokkos::CudaUVMSpace, Cabana::FullNeighborTag>, ExecParameters=<Kokkos::Cuda>]" 
/ccsopen/home/jgalarowicz/cabana/core/unit_test/tstNeighborList.hpp(383): here

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

6 errors detected in the compilation of "/tmp/tmpxft_00006d4a_00000000-6_tstSort_Cuda.cpp1.ii".
make[2]: *** [core/unit_test/CMakeFiles/Sort_test_Cuda.dir/Cuda/tstSort_Cuda.cpp.o] Error 1
/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

2 errors detected in the compilation of "/tmp/tmpxft_00006c44_00000000-6_tstSlice_Cuda.cpp1.ii".
make[2]: *** [core/unit_test/CMakeFiles/Slice_test_Cuda.dir/Cuda/tstSlice_Cuda.cpp.o] Error 1
/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2018): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelScan< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::initial const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2018): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2074): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelScan< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::final const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2074): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2087): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelScan< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::final const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(2087): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(839): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelReduce< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Max<int,  ::Kokkos::HostSpace> ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(839): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(685): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::TeamPolicy< ::Kokkos::Cuda,  ::Kokkos::Schedule< ::Kokkos::Dynamic>  > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(685): error: identifier " const" is undefined in device code

22 errors detected in the compilation of "/tmp/tmpxft_00006c19_00000000-6_tstNeighborList_Cuda.cpp1.ii".
make[2]: *** [core/unit_test/CMakeFiles/NeighborList_test_Cuda.dir/Cuda/tstNeighborList_Cuda.cpp.o] Error 1
/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: calling a __host__ function(" const") from a __device__ function("Kokkos::Impl::ParallelFor< ::,  ::Kokkos::RangePolicy< ::Kokkos::Cuda > ,  ::Kokkos::Cuda> ::operator () const") is not allowed

/ccsopen/home/jgalarowicz/kokkos.install/include/Cuda/Kokkos_Cuda_Parallel.hpp(474): error: identifier " const" is undefined in device code

6 errors detected in the compilation of "/tmp/tmpxft_00006d49_00000000-6_tstDeepCopy_Cuda.cpp1.ii".
make[2]: *** [core/unit_test/CMakeFiles/DeepCopy_test_Cuda.dir/Cuda/tstDeepCopy_Cuda.cpp.o] Error 1
make[2]: Target `core/unit_test/CMakeFiles/DeepCopy_test_Cuda.dir/build' not remade because of errors.
make[2]: Leaving directory `/autofs/nccsopen-svm1_home/jgalarowicz/cabana/build'
make[1]: *** [core/unit_test/CMakeFiles/DeepCopy_test_Cuda.dir/all] Error 2
make[2]: Target `core/unit_test/CMakeFiles/Slice_test_Cuda.dir/build' not remade because of errors.
make[2]: Leaving directory `/autofs/nccsopen-svm1_home/jgalarowicz/cabana/build'
make[1]: *** [core/unit_test/CMakeFiles/Slice_test_Cuda.dir/all] Error 2
make[2]: Target `core/unit_test/CMakeFiles/NeighborList_test_Cuda.dir/build' not remade because of errors.
make[2]: Leaving directory `/autofs/nccsopen-svm1_home/jgalarowicz/cabana/build'
make[1]: *** [core/unit_test/CMakeFiles/NeighborList_test_Cuda.dir/all] Error 2
make[2]: Target `core/unit_test/CMakeFiles/Tuple_test_Cuda.dir/build' not remade because of errors.
make[2]: Leaving directory `/autofs/nccsopen-svm1_home/jgalarowicz/cabana/build'
make[1]: *** [core/unit_test/CMakeFiles/Tuple_test_Cuda.dir/all] Error 2
[ 72%] Linking CXX executable CartesianGrid_test
```

Also, I had been running with Christoph's cabana cmake arguments, when I use ones in line with your script from above, I see these errors:
```
<pre>===== CMAKE_OPTS=Cuda
-- The CXX compiler identification is GNU 6.4.0
-- Check for working CXX compiler: /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper
-- Check for working CXX compiler: /ccsopen/home/jgalarowicz/kokkos.install/bin/nvcc_wrapper -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - failed
-- Found KOKKOS: /ccsopen/home/jgalarowicz/kokkos.install  
-- Found OpenMP_CXX: -fopenmp (found version &quot;4.5&quot;) 
-- Found OpenMP: TRUE (found version &quot;4.5&quot;)  
-- The CUDA compiler identification is NVIDIA 9.2.148
-- Check for working CUDA compiler: /sw/ascent/cuda/9.2.148/bin/nvcc
-- Check for working CUDA compiler: /sw/ascent/cuda/9.2.148/bin/nvcc -- works
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Enable Devices: Serial;OpenMP;Cuda
-- Found Doxygen: /usr/bin/doxygen (found version &quot;1.8.5&quot;) found components:  doxygen missing components:  dot
CMake Warning (dev) at CMakeLists.txt:69 (find_package):
  Policy CMP0074 is not set: find_package uses &lt;PackageName&gt;_ROOT variables.
  Run &quot;cmake --help-policy CMP0074&quot; for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Environment variable MPI_ROOT is set to:

    /autofs/nccsopen-svm1_sw/ascent/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.2.0.10-20181214-o4r7cptz5i4og4c5x6ngsacosjqeo6l7

  For compatibility, CMake is ignoring the variable.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found MPI_CXX: /autofs/nccsopen-svm1_sw/ascent/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.2.0.10-20181214-o4r7cptz5i4og4c5x6ngsacosjqeo6l7/lib/libmpiprofilesupport.so (found version &quot;3.1&quot;) 
-- Found MPI: TRUE (found version &quot;3.1&quot;)  
-- Performing Test COMPILER_SUPPORTS_MARCH
-- Performing Test COMPILER_SUPPORTS_MARCH - Failed
-- Found Git: /usr/bin/git (found version &quot;1.8.3.1&quot;) 
-- Cabana Revision = &apos;176db528ee2e4dc483aa58f99ad6440d3b913920&apos;
CMake Error at core/unit_test/CMakeLists.txt:59 (add_test):
  add_test given test NAME &quot;CommunicationPlan_test_Serial_1&quot; which already
  exists in this directory.


CMake Error at core/unit_test/CMakeLists.txt:59 (add_test):
  add_test given test NAME &quot;Distributor_test_Serial_1&quot; which already exists
  in this directory.


CMake Error at core/unit_test/CMakeLists.txt:59 (add_test):
  add_test given test NAME &quot;Halo_test_Serial_1&quot; which already exists in this
  directory.

</pre>
```
Do you know what might be causing these?

These were the cabana build options:
```
<pre>cd cabana
rm -rf build
mkdir build
cd build
cmake  .. \
	-DCMAKE_PREFIX_PATH=${CI_PROJECT_DIR}/kokkos.install \
        -DKOKKOS_SETTINGS_DIR=${CI_PROJECT_DIR}/kokkos.install \
        -DKOKKOS_LIBRARY=${CI_PROJECT_DIR}/kokkos.install/lib/libkokkos.a \
        -DKOKKOS_INCLUDE_DIR=${CI_PROJECT_DIR}/kokkos.install/include \
        -DCabana_ENABLE_Serial=ON \
        -DCabana_ENABLE_OpenMP=ON \
        -DCabana_ENABLE_Cuda=ON \
        -DCabana_ENABLE_MPI=ON \
        -DCabana_ENABLE_TESTING=ON \
        -DCabana_ENABLE_EXAMPLES=ON \
	${CMAKE_OPTS[@]} ${CMAKE_EXTRA_ARG}

</pre>
```
Thanks!

```
    56	        add_executable(${_test}_test_${_device} ${_device}/tst${_test}_${_device}.cpp mpi_unit_test_main.cpp)
    57	        target_link_libraries(${_test}_test_${_device} cabanacore cabana_core_gtest)
    58	        # foreach(_np 1 ${MPIEXEC_MAX_NUMPROCS})
    59	        list(APPEND TEST_MPIEXEC_NUMPROCS  1)
    60	        if (MPIEXEC_MAX_NUMPROCS GREATER 1)
    61	          list(APPEND TEST_MPIEXEC_NUMPROCS ${MPIEXEC_MAX_NUMPROCS})
    62	        endif()
    63	        foreach(_np ${TEST_MPIEXEC_NUMPROCS})
    64	          add_test(NAME ${_test}_test_${_device}_${_np} COMMAND
    65	            ${MPIEXEC} ${MPIEXEC_NUMPROC_FLAG} ${_np} ${_test}_test_${_device} --gtest_color=yes)
    66	        endforeach()
    67	      endforeach()
    68	    endif()
```
Now I'm seeing the error at line 64:
```
-- Found Git: /usr/bin/git (found version "1.8.3.1") 
-- Cabana Revision = '176db528ee2e4dc483aa58f99ad6440d3b913920'
CMake Error at core/unit_test/CMakeLists.txt:64 (add_test):
  add_test given test NAME "Distributor_test_Serial_1" which already exists
  in this directory.


CMake Error at core/unit_test/CMakeLists.txt:64 (add_test):
  add_test given test NAME "Halo_test_Serial_1" which already exists in this
  directory.


CMake Error at core/unit_test/CMakeLists.txt:64 (add_test):
  add_test given test NAME "Halo_test_Serial_1" which already exists in this
  directory.
```

All 59 tests pass now.   
```
Running tests...
Test project /ccsopen/home/jgalarowicz/cabana/build
      Start  1: Version_test
 1/59 Test  #1: Version_test ......................   Passed    1.97 sec
      Start  2: Index_test
 2/59 Test  #2: Index_test ........................   Passed    1.77 sec
      Start  3: CartesianGrid_test
 3/59 Test  #3: CartesianGrid_test ................   Passed    1.97 sec
      Start  4: SoA_test
 4/59 Test  #4: SoA_test ..........................   Passed    1.74 sec
      Start  5: AoSoA_test_Serial
 5/59 Test  #5: AoSoA_test_Serial .................   Passed    1.86 sec
...
...
<pre>56/59 Test #56: Core_tutorial_06 ..................   Passed    1.82 sec
      Start 57: Core_tutorial_10_openmp
57/59 Test #57: Core_tutorial_10_openmp ...........   Passed    1.77 sec
      Start 58: Core_tutorial_11
58/59 Test #58: Core_tutorial_11 ..................   Passed    1.91 sec
      Start 59: Core_tutorial_10_cuda
59/59 Test #59: Core_tutorial_10_cuda .............   Passed    1.91 sec

100% tests passed, 0 tests failed out of 59

Total Test time (real) = 118.39 sec
=============================================================
</pre>
```


The only error related things I'm seeing is this:
```
<pre>make[2]: Leaving directory `/autofs/nccsopen-svm1_home/jgalarowicz/cabana/build&apos;
[ 16%] Built target AdvancedCudaSlice
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp: In function &apos;void run()&apos;:
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;cx&apos; in &apos;asm&apos;
   asm volatile (&quot;rdtscp;shlq $32,%%rdx;orq %%rdx,%%rax;movq %%rax,%0&quot;:&quot;=q&quot;(u)::&quot;%rax&quot;, &quot;%rdx&quot;, &quot;%rcx&quot;);
                                                                                                        ^
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;dx&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;ax&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;cx&apos; in &apos;asm&apos;
   asm volatile (&quot;rdtscp;shlq $32,%%rdx;orq %%rdx,%%rax;movq %%rax,%0&quot;:&quot;=q&quot;(u)::&quot;%rax&quot;, &quot;%rdx&quot;, &quot;%rcx&quot;);
                                                                                                        ^
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;dx&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;ax&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;cx&apos; in &apos;asm&apos;
   asm volatile (&quot;rdtscp;shlq $32,%%rdx;orq %%rdx,%%rax;movq %%rax,%0&quot;:&quot;=q&quot;(u)::&quot;%rax&quot;, &quot;%rdx&quot;, &quot;%rcx&quot;);
                                                                                                        ^
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;dx&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;ax&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;cx&apos; in &apos;asm&apos;
   asm volatile (&quot;rdtscp;shlq $32,%%rdx;orq %%rdx,%%rax;movq %%rax,%0&quot;:&quot;=q&quot;(u)::&quot;%rax&quot;, &quot;%rdx&quot;, &quot;%rcx&quot;);
                                                                                                        ^
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;dx&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;ax&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;cx&apos; in &apos;asm&apos;
   asm volatile (&quot;rdtscp;shlq $32,%%rdx;orq %%rdx,%%rax;movq %%rax,%0&quot;:&quot;=q&quot;(u)::&quot;%rax&quot;, &quot;%rdx&quot;, &quot;%rcx&quot;);
                                                                                                        ^
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;dx&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;ax&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;cx&apos; in &apos;asm&apos;
   asm volatile (&quot;rdtscp;shlq $32,%%rdx;orq %%rdx,%%rax;movq %%rax,%0&quot;:&quot;=q&quot;(u)::&quot;%rax&quot;, &quot;%rdx&quot;, &quot;%rcx&quot;);
                                                                                                        ^
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;dx&apos; in &apos;asm&apos;
/ccsopen/home/jgalarowicz/cabana/core/example/benchmark/Cabana_peakflops.cpp:23:104: error: unknown register name &apos;ax&apos; in &apos;asm&apos;
make[2]: Leaving directory `/autofs/nccsopen-svm1_home/jgalarowicz/cabana/build&apos;
make[2]: *** [core/example/benchmark/CMakeFiles/PeakFlops.dir/Cabana_peakflops.cpp.o] Error 1
make[2]: Target `core/example/benchmark/CMakeFiles/PeakFlops.dir/build&apos; not remade because of errors.
make[2]: Leaving directory `/autofs/nccsopen-svm1_home/jgalarowicz/cabana/build&apos;
make[1]: *** [core/example/benchmark/CMakeFiles/PeakFlops.dir/all] Error 2
[ 16%] Built target ArrayOfStructsOfArrays
[ 17%] Linking CXX executable DeepCopy
</pre>
```

--- a/.ecp-gitlab-ci.yml
+++ b/.ecp-gitlab-ci.yml
@@ -1,25 +1,29 @@
 # see https://docs.gitlab.com/ce/ci/yaml/README.html for all available options
+
 variables:
-  CI_DEBUG_SERVICES: trace
-  SCHEDULER_PARAMETERS: "-P STF002 -J ginsburgtest -W 0:03 -nnodes 2"
+  SCHEDULER_PARAMETERS: "-J Cabana_CI -W 0:05 -nnodes 1 -P GEN007CITEST"
 
 stages:
   - buildKokkos
   - build
-#  - test
+# serialize testing to 2 per stage to working around bug in ci runner on ascent
+  - test
+  - test2
+  - test3
+  - test4
 
 .BuildKokkos:
   stage: buildKokkos
-  script:
-    - CI_PROJECT_DIR=${PWD}
+  before_script:
     - module load gcc
+  script:
     - j="$(grep -c processor /proc/cpuinfo 2>/dev/null)" || j=0; ((j++))
     - for i in ${BACKENDS}; do KOKKOS_OPTS+=( --with-${i,,[A-Z]} ); done
     - git clone --depth=1 https://github.com/kokkos/kokkos.git &&
       pushd kokkos &&
       mkdir build &&
       pushd build &&
-      ../generate_makefile.bash --prefix=${CI_PROJECT_DIR}/kokkos.install ${KOKKOS_OPTS[@]} &&
+      ../generate_makefile.bash --prefix=${CI_PROJECT_DIR}/kokkos.install ${KOKKOS_OPTS[@]} ${KOKKOS_EXTRA_ARGS} &&
       make -j${j} &&
       make install &&
       popd &&
@@ -32,20 +36,70 @@ stages:
 
 .Build:
   stage: build
-  script:
-    - CI_PROJECT_DIR=${PWD}
-    - module load cmake
+  before_script:
     - module load gcc
+  script:
+    - module load cmake
     - for i in ${BACKENDS}; do CMAKE_OPTS+=( -DCabana_ENABLE_${i}=ON ); done
     - j="$(grep -c processor /proc/cpuinfo 2>/dev/null)" || j=0; ((j++))
     - mkdir build && cd build &&
-      cmake -DCMAKE_PREFIX_PATH=${CI_PROJECT_DIR}/kokkos.install -DCabana_ENABLE_TESTING=ON -DCabana_ENABLE_Serial=OFF -DCabana_ENABLE_EXAMPLES=OFF ${CMAKE_OPTS[@]} .. &&
-      make -k -j${j} VERBOSE=1 &&
-      make test CTEST_OUTPUT_ON_FAILURE=1
+      cmake -DCMAKE_PREFIX_PATH=${CI_PROJECT_DIR}/kokkos.install -DCabana_ENABLE_TESTING=ON -DCabana_ENABLE_Serial=OFF -DCabana_ENABLE_EXAMPLES=ON ${CMAKE_OPTS[@]} ${CMAKE_EXTRA_ARGS} .. &&
+      make -k -j${j} VERBOSE=1
   tags:
     - nobatch
+  artifacts:
+    paths:
+      - build/
+
+.Test:
+  stage: test
+  before_script:
+    - module load gcc
+  script:
+    - cd build
+    - make test CTEST_OUTPUT_ON_FAILURE=1
+  tags:
+    - batch
+
+.BuildKokkos_Cuda:
+  extends: .BuildKokkos
+  variables:
+    KOKKOS_EXTRA_ARGS: "--arch=Volta70,Power9 --with-cuda-options=enable_lambda"
+  before_script:
+    - module load gcc/6.4.0
+    - module load cuda
+
+.Build_Cuda:
+  extends: .Build
+  variables:
+    CMAKE_EXTRA_ARGS: "-DCMAKE_CXX_COMPILER=${CI_PROJECT_DIR}/kokkos.install/bin/nvcc_wrapper"
+  before_script:
+    - module load gcc/6.4.0
+    - module load cuda
+
+.Test_Cuda:
+  extends: .Test
+  before_script:
+    - module load gcc/6.4.0
+    - module load cuda
+
+BuildKokkos Cuda:
+  variables:
+    BACKENDS: "Cuda"
+  extends: .BuildKokkos_Cuda
+
+Build Cuda:
+  variables:
+    BACKENDS: "Cuda"
+  extends: .Build_Cuda
   dependencies:
-   - BuildKokkos
+   - BuildKokkos Cuda
+
+Test Cuda:
+  extends: .Test_Cuda
+  dependencies:
+   - Build Cuda
+   - BuildKokkos Cuda
 
 BuildKokkos Serial:
   variables:
@@ -57,6 +111,12 @@ Build Serial:
     BACKENDS: "Serial"
   extends: .Build
   dependencies:
+   - BuildKokkos Serial
+
+Test Serial:
+  extends: .Test
+  dependencies:
+   - Build Serial
    - BuildKokkos Serial
 
 BuildKokkos OpenMP:
@@ -71,6 +131,13 @@ Build OpenMP:
   dependencies:
    - BuildKokkos OpenMP
 
+Test OpenMP:
+  extends: .Test
+  stage: test2
+  dependencies:
+   - Build OpenMP
+   - BuildKokkos OpenMP
+
 BuildKokkos Pthread:
   variables:
     BACKENDS: "Pthread"
@@ -81,6 +148,13 @@ Build Pthread:
     BACKENDS: "Pthread"
   extends: .Build
   dependencies:
+   - BuildKokkos Pthread
+
+Test Pthread:
+  extends: .Test
+  stage: test2
+  dependencies:
+   - Build Pthread
    - BuildKokkos Pthread
 
 BuildKokkos Serial Pthread:
@@ -95,6 +169,13 @@ Build Serial Pthread:
   dependencies:
    - BuildKokkos Serial Pthread
 
+Test Serial Pthread:
+  extends: .Test
+  stage: test3
+  dependencies:
+   - Build Serial Pthread
+   - BuildKokkos Serial Pthread
+
 BuildKokkos Serial OpenMP:
   variables:
     BACKENDS: "Serial OpenMP"
@@ -106,3 +187,49 @@ Build Serial OpenMP:
   extends: .Build
   dependencies:
    - BuildKokkos Serial OpenMP
+
+Test Serial OpenMP:
+  extends: .Test
+  stage: test3
+  dependencies:
+   - Build Serial OpenMP
+   - BuildKokkos Serial OpenMP
+
+BuildKokkos Serial Cuda:
+  variables:
+    BACKENDS: "Serial Cuda"
+  extends: .BuildKokkos_Cuda
+
+Build Serial Cuda:
+  variables:
+    BACKENDS: "Serial Cuda"
+  extends: .Build_Cuda
+  dependencies:
+   - BuildKokkos Serial Cuda
+
+Test Serial Cuda:
+  extends: .Test_Cuda
+  stage: test4
+  dependencies:
+   - Build Serial Cuda
+   - BuildKokkos Serial Cuda
+
+BuildKokkos Serial Cuda OpenMP:
+  variables:
+    BACKENDS: "Serial Cuda OpenMP"
+  extends: .BuildKokkos_Cuda
+
+Build Serial Cuda OpenMP:
+  variables:
+    BACKENDS: "Serial Cuda OpenMP"
+  extends: .Build_Cuda
+  dependencies:
+   - BuildKokkos Serial Cuda OpenMP
+
+Test Serial Cuda OpenMP:
+  extends: .Test_Cuda
+  stage: test4
+  dependencies:
+   - Build Serial Cuda OpenMP
+   - BuildKokkos Serial Cuda OpenMP
+

--- a/.ecp-gitlab-ci.yml
+++ b/.ecp-gitlab-ci.yml
@@ -190,7 +190,7 @@ Build Serial OpenMP:
 
 Test Serial OpenMP:
   extends: .Test
-  stage: test3
+  stage: test4
   dependencies:
    - Build Serial OpenMP
    - BuildKokkos Serial OpenMP
@@ -209,7 +209,7 @@ Build Serial Cuda:
 
 Test Serial Cuda:
   extends: .Test_Cuda
-  stage: test4
+  stage: test3
   dependencies:
    - Build Serial Cuda
    - BuildKokkos Serial Cuda

